README.md

```diff
@@ -1,24 +1,220 @@
 
-* Database initialization
+Backend is implement as API-only. The front end is implemented utilizing the React framework.
 
-* How to run the test suite
+-  The backend Rails app is implemented as API-only. The application starts by fetching the `users` method on the backend, if the backend is available then the user will login by making a query to `users/new_session`.
 
-* Services (job queues, cache servers, search engines, etc.)
+-  The room controller provides an `index` method to get all the data for rooms, and a `show` method to get details about a specific room. The `destroy` method deactivates the room when the user sends a delete query by clicking on delete.
 
-* Deployment instructions
+-  The reservation controller also provides `index` methods. Index is used to get a list of all reservations made. 
+ 
+## 🛠 Built With <a id="built-with"></a>
 
-* ...
+### Tech Stack <a id="tech-stack"></a>
+
+<details>
+  <summary>Ruby on Rails</summary>
+  <ul>
+    <li><a>https://rubyonrails.org/</a></li>   
+  </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a id="key-features"></a>
+
+- **Ability to see the featured rooms on the homepage**
+- **The ability to reserve rooms based on selecting the date and room name**
+- **The ability to add rooms to the database**
+- **The ability to delete rooms**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a id="live-demo"></a>
+
+Live demo for the website [Coming Soon]().
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🎬 Kanban Board <a id="kanban"></a>
+
+Link to the [Kanban Board](https://github.com/ProgramKingAbel/hotel-hub-backend/projects/1)
+
+All 5 authors have been contributing to this project from start to finish.
+
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a id="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project, you need to:
+
+- Install Ruby
+- Install Rails
+- Install Posgresql
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Setup
+
+Clone this repository to your desired folder:
+
+``` 
+  git clone https://github.com/ProgramKingAbel/hotel-hub-backend.git 
+```
+
+### Install
+
+Install this project with these commands:
+
+```
+  cd hotel-hub
+  bundle install
+  rails db:create
+  rails db:migrate
+  rails db:seed
+  ```
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Usage
+
+
+### Install
+ 
+ ```cd hotel-hub-frontend
+    npm install
+ ```
+ <p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Usage
+ 
+To run the project execute the following command:
+
+``` 
+ rails s
+```
+
+or
+
+```
+./bin/dev
+```
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+###  Tests <a id="run-tests"></a>
+
+To run the tests, execute the following command:
+
+```
+rspec spec
```

Comment:
- [ ] When I run the tests according to instructions, 2 tests are failing:

![image](https://github.com/ProgramKingAbel/hotel-hub-backend/assets/57421823/0c004493-5794-4661-be61-e8fc1a945c2b)

This is because those tests are rswag tests, and rspec won't run them correctly.

## Solution

Kindly, be more specific in your instructions:

```
rspec ./spec/models
rspec ./spec/requests/api/v1
```

Result:

![image](https://github.com/ProgramKingAbel/hotel-hub-backend/assets/57421823/2d46dc17-073b-4e84-a856-17c9f0a695df)

That way, none of your tests will fail. 👍